### PR TITLE
feat(Bonus Pagamenti Digitali): [#176578578] Cobadge card preview

### PR DESCRIPTION
**Pull request by debiff**
## Short description
This PR add the preview for the cobadge card

## List of changes proposed in this pull request
- Added selector for cobadge card
- Enhanced CreditCardPaymentMethod with the optional abiInfo
- Added the cobadge preview card

https://user-images.githubusercontent.com/11773070/105532404-112a8000-5ceb-11eb-8e17-65774c120fb3.mov

### CODE REVIEW

File: ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps> &
+  OwnProps;
+
+const BASE_IMG_W = 160;
+const BASE_IMG_H = 20;
+/**
+ * Render the image (if available) or the bank name (if available)
+ * or the generic bancomatPay string (final fallback).
```
Comment (fabriziofff) [inline ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx:37]:
```suggestion
 * or the `CreditCardPaymentMethod caption` string (final fallback).
```

File: ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    return cardMapIcon[imageName] ?? defaultCardIcon;
+  });
+};
+/**
+ * A card preview for a bancomat card
+ * @param props
+ * @constructor
+ */
+const BPayWalletPreview: React.FunctionComponent<Props> = props => {
```
Comment (fabriziofff) [inline ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx:78]:
```suggestion
const CoBadgeWalletPreview: React.FunctionComponent<Props> = props => {
```

File: ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+};
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  navigateToCobadgeDetails: (cobadge: CreditCardPaymentMethod) =>
+    dispatch(navigateToCobadgeDetailScreen(cobadge))
+});
+
+const mapStateToProps = (_: GlobalState) => ({});
+
+export default connect(mapStateToProps, mapDispatchToProps)(BPayWalletPreview);
```
Comment (fabriziofff) [inline ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx:100]:
```suggestion
export default connect(mapStateToProps, mapDispatchToProps)(CoBadgeWalletPreview);
```

File: ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  const brandLogo = cobadge.info.brandLogo;
+
+  return fromNullable(brandLogo).fold(defaultCardIcon, bL => {
+    const imageName = getResourceNameFromUrl(bL);
+
+    return cardMapIcon[imageName] ?? defaultCardIcon;
+  });
+};
+/**
+ * A card preview for a bancomat card
```
Comment (fabriziofff) [inline ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx:74]:
```suggestion
 * A card preview for a co-badge card
```

File: ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        <Image
+          source={{ uri: imageUrl }}
+          style={imageStyle}
+          testID={"bankLogo"}
+        />
+      ) : null;
+    }
+  );
+
+const getBrandLogo = (cobadge: CreditCardPaymentMethod) => {
```
Comment (fabriziofff) [inline ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx:64]: In order to have a single source of truth, what you think to use the function `getCardIconFromBrandLogo` that return the card icon starting from `CardInfo`?
https://github.com/pagopa/io-app/blob/80fc100b65e5eca03142ff6c6a1d1875d7f09166/ts/components/wallet/card/Logo.tsx#L45

**Review by fabriziofff — CHANGES_REQUESTED** [review]

File: ts/store/reducers/wallet/wallets.ts [hunk 6]
```diff
@@ -266,6 +266,17 @@ export const bPayListVisibleInWalletSelector = createSelector(
...
+/**
+ * Return a CoBadge list visible in the wallet
+ */
+export const cobadgeListVisibleInWalletSelector = createSelector(
+  [creditCardListVisibleInWalletSelector],
+  (creditCardListPot): pot.Pot<ReadonlyArray<CreditCardPaymentMethod>, Error> =>
+    pot.map(creditCardListPot, creditCardList =>
+      creditCardList.filter(cc => cc.pagoPA === false)
+    )
+);
```
Comment (fabriziofff) [inline ts/store/reducers/wallet/wallets.ts:278]: As discussed in https://www.pivotaltracker.com/story/show/176642752, we should also add the condition `issuerAbiCode !== undefined`
```suggestion
export const cobadgeListVisibleInWalletSelector = createSelector(
  [creditCardListVisibleInWalletSelector],
  (creditCardListPot): pot.Pot<ReadonlyArray<CreditCardPaymentMethod>, Error> =>
    pot.map(creditCardListPot, creditCardList =>
            creditCardList.filter(
        cc => cc.pagoPA === false && cc.info.issuerAbiCode !== undefined
      )
    )
);
```

**Review by fabriziofff — CHANGES_REQUESTED** [review]

File: ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx (hunk 1 above)
Comment (debiff) [reply ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx]: Solved on [4087a27](https://github.com/pagopa/io-app/pull/2746/commits/4087a272de2793511ac196a69403f70de80fa331)

File: ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx (hunk 2 above)
Comment (debiff) [reply ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx]: Solved on [4087a27](https://github.com/pagopa/io-app/pull/2746/commits/4087a272de2793511ac196a69403f70de80fa331)

File: ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx (hunk 3 above)
Comment (debiff) [reply ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx]: Solved on [4087a27](https://github.com/pagopa/io-app/pull/2746/commits/4087a272de2793511ac196a69403f70de80fa331)

File: ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx (hunk 4 above)
Comment (debiff) [reply ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx]: Solved on [4087a27](https://github.com/pagopa/io-app/pull/2746/commits/4087a272de2793511ac196a69403f70de80fa331)

File: ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx (hunk 5 above)
Comment (debiff) [reply ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx]: Sure, thank you. I didn't see this function!
Solved on [37a4d42](https://github.com/pagopa/io-app/pull/2746/commits/37a4d42bc6556267d97c13c65daffb30293eba98)

File: ts/store/reducers/wallet/wallets.ts (hunk 6 above)
Comment (debiff) [reply ts/store/reducers/wallet/wallets.ts]: Solved on [ff295e4](https://github.com/pagopa/io-app/pull/2746/commits/ff295e42f7fee1a25fb674772d3df97846c2aa85)

**Review by fabriziofff — APPROVED** [review]

**Pull request merged**